### PR TITLE
chore: fix mintPayment type

### PR DESCRIPTION
**Pull request by turadg**
## Description

The removal of `MintPayment` type wasn't complete. (h/t @mhofman https://github.com/Agoric/agoric-sdk/pull/4736/files#r830781495) I wonder what it would take for the type system to have noted this problem. I suppose tsc is permissive with type annotations in JSdoc because they could be for another type specification system. 

I didn't restore the callback because it's not a callback/interface. It's a function that exists only on `Mint` objects so I corrected the type inline. 

(While iterating on approaches I defined the `PaymentLedger` type and it's left in here)

### Security Considerations

--

### Documentation Considerations

--

### Testing Considerations

--

### CODE REVIEW

**Comment by mhofman** [comment]
> I wonder what it would take for the type system to have noted this problem.

`noImplicitAny` is what it would take

File: packages/ERTP/src/paymentLedger.js [hunk 1]
```diff
@@ -365,3 +365,5 @@ export const makePaymentLedger = (
     mint,
   });
 };
+
+/** @typedef {ReturnType<makePaymentLedger>} PaymentLedger */
```
Comment (mhofman) [inline packages/ERTP/src/paymentLedger.js:369]: Drive-by changes?

**Review by mhofman — APPROVED** [review]
LGTM, but I can't find any about the paymentLedger changes in the description

File: packages/ERTP/src/paymentLedger.js (hunk 1 above)
Comment (turadg) [reply packages/ERTP/src/paymentLedger.js:369]: yes. I'll add to description

**Comment by turadg** [comment]
>> I wonder what it would take for the type system to have noted this problem.

> `noImplicitAny` is what it would take

I'm not sure of that. Here's a [playground with all typechecking options disabled](https://www.typescriptlang.org/play?strict=false&noImplicitAny=false&strictFunctionTypes=false&strictPropertyInitialization=false&strictBindCallApply=false&noImplicitThis=false&noImplicitReturns=false&alwaysStrict=false&target=99&jsx=0&filetype=js#code/PQKhCgAIUgBAXAngBwKYBNUDNIG8D2ARgFaoDG8AvpAGL75QyzIBO+aLSeAKiqgHL54AEWwBLAHYZqhAIYtGwcOFBMkaPHXzUQSsvgkBneJCz1IAXjxyWALkgBGSgG5wQA) and it still can report that a type is being used that is not defined.
Screen Shot 2022-03-21 at 9 13 36 AM

So something is different about our setup that `MintPayment` didn't error and it's not `noImplicitAny`.
